src/SobelNode.js

```diff
@@ -0,0 +1,28 @@
+(function(TextureGen) {
+  'use strict';
+
+  class SobelNode extends TextureGen.CanvasNode {
+    constructor(id) {
+      super(id, 'Sobel', ['Image']);
```

Comment:
Since this pull request was created, the format for defining inputs has been changed. The super call would need to change to:

```
super(id, 'Sobel', [new TextureGen.GraphInput({name: 'Image'})]);
```